enacts/calc.py

```diff
@@ -594,6 +594,423 @@ def _cess_date(dry_thresh, dry_spell_length_thresh, sm_func, time_coord):
+    interval_dim : str
+        name of pd.Interval dimension to be replaced
+    to_point : str, optional
+        "left", "mid" or "right" point of `interval_dim` intervals
+        default is "mid"
+    keep_attrs : boolean, optional
+        keep attributes from `interval_dim` to replacing point-wise dimension
+        default is True
+
+    Returns
+    -------
+    point_data : xr.DataArray or xr.Dataset
+        of whicn interval dimension has been replaced by point dimension
+
+    See Also
+    --------
+    pandas.Interval, interval_to_point, xarray.assign_coords, xarray.swap_dims
+    """
+    point_dim = interval_to_point(
+        interval_data[interval_dim], to_point=to_point, keep_attrs=keep_attrs
+    )
+    return (
+        interval_data
+        .assign_coords({point_dim.name : (interval_dim, point_dim.data)})
+        .swap_dims({interval_dim: point_dim.name})
+    )
+
+
+def groupby_dekads(daily_data, time_dim="T"):
+    """ Groups `daily_data` by dekads for grouping operations
+
+    Parameters
+    ----------
+    daily_data : xr.DataArray or xr.Dataset
+        daily data
+    time_dim : str, optional
+        name of daily time dimenstion, default is "T"
+
+    Returns
+    -------
+    grouped : xr.DataArrayGroupBy or xr.DatasetGroupBy
+        `daily_data` grouped by dekads
+
+    See Also
+    --------
+    xarray.groupby_bins, xarray.DataArrayGroupBy, xarray.DatasetGroupBy
+    """
+    # dekad bins are located at midnight
+    dekad_edges = pd.date_range(
+        start=daily_data[time_dim][0].dt.floor("D").values,
+        end=(daily_data[time_dim][-1] + np.timedelta64(1, "D")).dt.floor("D").values,
+        freq="1D",
+    )
+    dekad_edges = dekad_edges.where(
+        (dekad_edges.day == 1) | (dekad_edges.day == 11) | (dekad_edges.day == 21)
+    ).dropna()
+    assert dekad_edges.size > 1, (
+        "daily_data must span at least one full dekad (need 2 edges to form 1 bin)"
+    )
+    return daily_data.groupby_bins(daily_data[time_dim], dekad_edges, right=False)
+
+
+def daily_to_dekad(daily_data, method=None, method_kwargs={}, time_dim="T"):
+    """ Reduce daily to dekad data
+
+    Parameters
+    ----------
+    daily_data : xr.DataArray or xr.Dataset
+        daily data
+    method : str, optional
+        xr.groupby_bins method to reduce `daily_data`
+        Default is None in which case a DataAttayGroupBy or DatasetGroupBy object is
+        returned
+    method_kwargs : dict(kwargs), optional
+        a dictionary of keyword arguments to pass to `method`
+        Default is an empty dict thus applying method default arguments
+    time_dim : str, optional
+        name of daily time dimenstion, default is "T"
+
+    Returns
+    -------
+    dekad_data : xr.DataArray or xr.Dataset or DataArrayGroupBy or DatasetGroupBy objects
+        daily data grouped by dekad if no `method` provided. Otherwise daily data
+        reduced to dekadal data according to `method`
```

Comment:
This function seems complicated and unnecessary. It's no harder to write
```
groupby_dekads(daily_data).sum()
```
than to write
```
daily_to_dekad(daily_data, method="sum")
```

Reply:
exit daily_to_dekad